src/libs/actions/IOU.ts

```diff
@@ -4673,7 +4673,10 @@ function hasIOUToApproveOrPay(chatReport: OnyxEntry<OnyxTypes.Report> | EmptyObj
 function approveMoneyRequest(expenseReport: OnyxTypes.Report | EmptyObject, full?: boolean) {
     const currentNextStep = allNextSteps[`${ONYXKEYS.COLLECTION.NEXT_STEP}${expenseReport.reportID}`] ?? null;
     let total = expenseReport.total ?? 0;
-    if (ReportUtils.hasHeldExpenses(expenseReport.reportID) && !full && !!expenseReport.unheldTotal) {
+    const transactions = TransactionUtils.getAllReportTransactions(expenseReport.reportID);
+    const hasHeldExpenses = transactions.some((transaction) => TransactionUtils.isOnHold(transaction));
```

Comment:
Because ReportUtils.hasHeldExpenses also get all transaction and use it to check the held request and we also use allTransactions here to get the heldTransactions, so I not use ReportUtils.hasHeldExpenses to prevent call `getAllReportTransactions` two times.

Reply:
@hoangzinh I updated.